nautobot_secrets_providers/providers/thycotic.py

```diff
@@ -0,0 +1,181 @@
+        is_valid_domain = "domain" in thycotic_settings and thycotic_settings["domain"] != ""
+        is_valid_password = "password" in thycotic_settings and thycotic_settings["password"] != ""
+        is_valid_tenant = "tenant" in thycotic_settings and thycotic_settings["tenant"] != ""
+        is_valid_token = "token" in thycotic_settings and thycotic_settings["token"] != ""
+        is_valid_username = "username" in thycotic_settings and thycotic_settings["username"] != ""
+        if_valid_ca_bundle_path = "ca_bundle_path" in thycotic_settings and thycotic_settings["ca_bundle_path"] != ""
+
+        return cls.query_thycotic_secret_server(
+            secret=secret,
+            base_url=thycotic_settings["base_url"],
+            ca_bundle_path=thycotic_settings["ca_bundle_path"] if if_valid_ca_bundle_path else None,
+            cloud_based=thycotic_settings["cloud_based"],
+            domain=thycotic_settings["domain"] if is_valid_domain else None,
+            password=thycotic_settings["password"] if is_valid_password else None,
+            secret_id=secret_id,
+            secret_selected_value=secret_selected_value,
+            tenant=thycotic_settings["tenant"] if is_valid_tenant else None,
+            token=thycotic_settings["token"] if is_valid_token else None,
+            username=thycotic_settings["username"] if is_valid_username else None,
+            caller_class=cls,
+        )
+
+    @staticmethod
+    def query_thycotic_secret_server(
+        secret,
+        base_url,
+        ca_bundle_path=None,
+        cloud_based=None,
+        domain=None,
+        password=None,
+        secret_id=None,
+        secret_selected_value=None,
+        tenant=None,
+        token=None,
+        username=None,
+        caller_class=None,
+    ):
+        """Query Thycotic Secret Server"""
+        # Ensure required parameters are set
+        if (token is None and (username is None or password is None)) or (
+            cloud_based and (tenant is None or username is None or password is None)
+        ):
+            raise exceptions.SecretProviderError(secret, caller_class, "Thycotic Secret Server is not configured!")
+
+        if ca_bundle_path is not None:
+            # Ensure cerificates file exists if ca_bundle_path is defined
+            if not Path(ca_bundle_path).exists():
+                raise exceptions.SecretProviderError(
+                    secret,
+                    caller_class,
+                    (
+                        "Thycotic Secret Server is not configured properly! "
+                        "Trusted certificates file not found: "
+                        "Environment variable 'REQUESTS_CA_BUNDLE': "
+                        f"{ca_bundle_path}."
+                    ),
+                )
+            from_env = os.getenv("REQUESTS_CA_BUNDLE", "")
+            if from_env != ca_bundle_path:
+                os.environ["REQUESTS_CA_BUNDLE"] = ca_bundle_path
```

Comment:
This feels like it could have unintended consequences to usage of `requests` elsewhere in this plugin or any other Nautobot plugin. Maybe we should have a `try... finally` block that restores the original value of `$REQUESTS_CA_BUNDLE` when this function is done?

Reply:
I commited the change now